source/Halibut/Transport/SecureListener.cs

```diff
@@ -118,49 +115,74 @@ public void Disconnect(string thumbprint)
 
         void Accept()
         {
-            var numberOfFailedAttemptsInRow = 0;
-
-            while (!cts.IsCancellationRequested)
-            {
-                try
+            // See: https://github.com/OctopusDeploy/Issues/issues/6035
+            // See: https://github.com/dotnet/corefx/issues/26034
+            var tokenCancellationListener = IsWindows()
+                ? cts.Token.Register(listener.Stop)
+                : (IDisposable) new NoopDisposable();
+            
+            var shouldAbort = IsWindows()
```

Comment:
`shouldAbort` doesn't tell me much about what this code does. Also the use of the ternary to return funcs is a bit confusing. I think this could be better written:

```
bool WaitForAnIncommingRequestIfNotWindows()
{
    if (IsWindows())
        return true;
    
    SpinWait.SpinUntil(() => cts.IsCancellationRequested || listener.Pending());
    return !cts.IsCancellationRequested;
}
```